src/tyr/actor.cc

```diff
@@ -46,15 +46,15 @@ std::string actor_t::route(const std::string& request_str, const std::function<v
   // set the interrupts
   pimpl->set_interrupts(interrupt);
   // parse the request
-  valhalla_request_t request;
-  request.parse(request_str, DirectionsOptions::route);
+  Api request;
```

Comment:
everything here in the external api becomes much simpler. you just pass the request from one stage to the next stage of the pipeline.